bundles/binding/org.openhab.binding.rwesmarthome/src/main/java/org/openhab/binding/rwesmarthome/internal/communicator/RWESmarthomeCommunicator.java

```diff
@@ -0,0 +1,346 @@
+			
+			} else if(LogicalDevice.Type_GenericActuator_Value.equals(ld.getType())) {
+				validParams = "variable";
+				exampleConfig.add("Switch\trweVariable" + ld.getId().substring(30) + "\t\"Variable " + ld.getName() + "\"\t<switch>\t(rwe) {" + context.getBindingType() + "=\"id=" + ld.getId() + ",param=variable\"}\n");
+				
+
+			} else if(LogicalDevice.Type_ThermostatActuator.equals(ld.getType())) {
+				continue; // ignore
+			} else {
+				validParams = "UNKNOWN";
+			}
+			
+			if("UNKNOWN".equals(validParams)) 
+				logger.debug("Found unsupported {} {} ({}/{}).", ld.getType(), ld.getId(), ld.getLocation().getName(), ld.getName());
+			else
+				logger.info("Found {} {} ({}/{}). Valid params: {}", ld.getType(), ld.getId(), ld.getLocation().getName(), ld.getName(), validParams);
+		}
+		
+		Collections.sort(exampleConfig);
+		exampleConfig.add(0, "Group rwe\n");
+		logger.info("Example configuration for RWE Smarthome items:\n" + StringUtils.join(exampleConfig, ""));
+	}
+	
+	/**
+	 * Returns true, if the communicator is already running and false, if not.
+	 * 
+	 * @return boolean
+	 */
+	public boolean isRunning() {
+		return rweSmarthomeSession == null;
+	}
+	
+	/**
+	 * load the state of the logical devices from RWE Smarthome SHC.
+	 * 
+	 * This is only necessary, when the RWE binding is started and the states unknown or
+	 * if the binding configuration has changed. Standard device state changes are
+	 * propagated using poll().
+	 */
+	public void loadDeviceStates() {
+		try {
+			rweSmarthomeSession.refreshLogicalDeviceStates();
+			logger.info("LogicalDevice states refreshed.");
+		} catch (RWESmarthomeSessionExpiredException e) {
+			logger.warn("Session expired!" + e.getMessage());
+
+			try {
+				rweSmarthomeSession.logon(context.getConfig().getUsername(), context.getConfig().getPassword(), context.getConfig().getHost());
+				logger.info("Login successful.");
+				
+				// trigger refresh again as we had to login again...
+				context.setBindingChanged(true);
+			} catch (Exception e1) {
+				logger.error("Could not recover session with RWE Smarthome: " + e1.getMessage(), e1);
+				stop();
+			}
+		}
+	}
+
+	/**
+	 * Polls for notifications and handles
+	 */
+	public void poll() {
+	
+		// Exit, if there is no valid session id
+		if (!context.getRweSmarthomeSession().isValid())
+			return;
+		
+		
+		try {
+			rweSmarthomeSession.getNotifications();
+			lastEventTime = System.currentTimeMillis();
+			
+		} catch (LogoutNotificationException e) {
+			// restart communicator to get back a valid session
+			logger.info("Session expired!" + e.getMessage());
+			stop();
+			start();
+			
+		} catch (RWESmarthomeSessionExpiredException e) {
+			logger.warn("Session expired!" + e.getMessage());
+
+			try {
+				rweSmarthomeSession.logon(context.getConfig().getUsername(), context.getConfig().getPassword(), context.getConfig().getHost());
+			} catch (LoginFailedException e1) {
+				logger.error("Error loging in: " + e1.getMessage());
```

Comment:
@teichsta, why do you suppose decreasing to INFO or DEBUG? RWE drops the session every six hours and the re-login is a normal case. But if does not work, there must be an error like: user/pass/host changed, host not reachable...
So I would say: error is fine here - isn't it?